### PR TITLE
Add Dark Mode user prefers & other a11y fixes ⌨️♿️

### PR DESCRIPTION
**Pull request by dmlb**
## Describe your changes
- add tailwind classes for dark mode
- clear all contrast errors
- add darkColor for tags (schema, interfaces, yml, template)
- show keyboard focus on entire tag (label and input wrapper)
- fix label for/id pair break when apostrophe
- add SR only new tab warnings
- add keyboard functionality to back to top button
- add more structural HTML elements (ol, nav)

Screen Shot 2023-06-19 at 6 19 42 PM
Screen Shot 2023-06-19 at 6 21 08 PM

## Related issue number/link
Add keyboard support related to recent merge https://github.com/ClimateTown/knowledge-hub/pull/223
contrast fixes for text:background 5:1 related to https://github.com/ClimateTown/knowledge-hub/issues/70#issuecomment-1552179812

### CODE REVIEW

File: src/routes/resources/+page.svelte [hunk 1]
```diff
@@ -271,10 +285,22 @@
...
+</ol>
+<div class="italic text-center m-4">Those are all the resources!</div>
+
 <button
-  class="w-10 h-10 inline-flex items-center justify-center rounded-full bg-green-500 text-white cursor-pointer fixed ease-in-out bottom-10 right-10 z-50"
+  class="w-10 h-10 inline-flex items-center justify-center rounded-full bg-green-700 dark:bg-green-900/75 text-white cursor-pointer fixed transition-opacity bottom-10 right-10 z-50 outline-offset-2 {showButton
+    ? 'opacity-100'
+    : 'opacity-0'}"
   on:click={scrollToTop}
-  class:hidden={!showButton}
```
Comment (VeckoTheGecko) [inline src/routes/resources/+page.svelte:277]: What was the motivation for this specific refactor? It still works, so no need to revert, just curious

**Review by VeckoTheGecko — APPROVED** [review]
Took me a wihle to wrap my head around everything, but after some sleuthing I agree with it all.

Colossal effort @dmlb !! A lot of attention put into the look of the site which I'm sure everyone will appreciate. I particularly like the YT feed and resource button animations on the home page, very satisfying 😄

**Comment by VeckoTheGecko** [comment]
Do you know if it would be easy to add a manual dark/light theme switcher in the website (kinda like how the [`pandas` docs](https://pandas.pydata.org/docs/) has a system, light, and dark preference).

_No need to implement_ mainly just curious to see you think its easy/possible. If easy, I can create a low priority issue for it.

**Comment by dmlb** [comment]
> Do you know if it would be easy to add a manual dark/light theme switcher in the website (kinda like how the [`pandas` docs](https://pandas.pydata.org/docs/) has a system, light, and dark preference).
> 
> _No need to implement_ mainly just curious to see you think its easy/possible. If easy, I can create a low priority issue for it.

It's doable for sure, need a little planning :)

**Pull request merged**